From d06844826999b2306dec61745acd8d7f8b51dba1 Mon Sep 17 00:00:00 2001
From: Atin Mukherjee <amukherj@redhat.com>
Date: Mon, 28 Mar 2016 10:49:43 +0530
Subject: [PATCH 36/80] libglusterfs: open cmd_history log file with O_APPEND and O_WRONLY

Backport of http://review.gluster.org/13829
            http://review.gluster.org/13847

Commit 8fdfa0c introduced a fix to ensure cmd_history file is log rotated
properly. However with this fix fdopen() is called with mode "a" on a fd which
was not opened with O_WRONLY & O_APPEND resulting into a fdopen() failure.

Fix is to open cmd_history.log file with O_CREATE|O_WRONLY|O_APPEND mode

Change-Id: I75ef350560aa6d5435c78c5fd83adfde1a73bfc3
BUG: 1321509
Signed-off-by: Atin Mukherjee <amukherj@redhat.com>
Reviewed-on: http://review.gluster.org/13829
Smoke: Gluster Build System <jenkins@build.gluster.com>
Reviewed-by: Niels de Vos <ndevos@redhat.com>
NetBSD-regression: NetBSD Build System <jenkins@build.gluster.org>
CentOS-regression: Gluster Build System <jenkins@build.gluster.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/71161
Tested-by: Niels de Vos <ndevos@redhat.com>
---
 libglusterfs/src/logging.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libglusterfs/src/logging.c b/libglusterfs/src/logging.c
index 0d44803..04b44b8 100644
--- a/libglusterfs/src/logging.c
+++ b/libglusterfs/src/logging.c
@@ -2443,7 +2443,7 @@ gf_cmd_log (const char *domain, const char *fmt, ...)
                 }
 
                 fd = open (ctx->log.cmd_log_filename,
-                           O_CREAT | O_RDONLY, S_IRUSR | S_IWUSR);
+                           O_CREAT | O_WRONLY | O_APPEND, S_IRUSR | S_IWUSR);
                 if (fd < 0) {
                         gf_msg (THIS->name, GF_LOG_CRITICAL, errno,
                                 LG_MSG_FILE_OP_FAILED, "failed to open "
-- 
1.7.1

